plugin/completion.py

```diff
@@ -41,10 +42,72 @@
 
 
 class LspResolveDocsCommand(sublime_plugin.TextCommand):
+    def run(self, edit: sublime.Edit, index: int) -> None:
+        item = CompletionHandler.completions[index]
+        detail = minihtml(item.get('detail') or "", self.view)
+        documentation = minihtml(item.get("documentation") or "", self.view)
+
+        # don't show the detail in the cooperate AC popup if it is already shown in the AC details filed.
+        self.is_detail_shown = bool(detail)
+
+        content = self.get_content(documentation, detail)
+        self.show_popup(content)
+
+        if not detail or not documentation:
+            # To make sure that the detail or documentation fields doesn't exist we need to resove the completion item.
+            # If those fields appear after the item is resolved we show them in the popup.
+            self.do_resolve(item)
+
+    def get_content(self, documentation: str, detail: str) -> str:
+        content = ""
+        if detail and not self.is_detail_shown:
+            content += "<div class='highlight' style='margin: 4px'>{}</div>".format(detail)
+
+        if documentation:
+            content += "<div style='margin: 10px'>{}</div>".format(documentation)
+
+        return content
+
+    def show_popup(self, content: str) -> None:
+        mdpopups.show_popup(
+            self.view,
+            content,
+            flags=sublime.COOPERATE_WITH_AUTO_COMPLETE,
+            max_width=480,
+            max_height=300,
+            allow_code_wrap=True
+        )
+
+    def do_resolve(self, item: dict) -> None:
+        session = session_for_view(self.view, 'completionProvider', self.view.sel()[0].begin())
+        if not session:
+            return
 
-    def run(self, edit: sublime.Edit) -> None:
-        self.view.show_popup('<div style="padding: 10px;">Showing documentation is under construction</div>',
-                             sublime.COOPERATE_WITH_AUTO_COMPLETE)
+        client = client_from_session(session)
+        if not client:
+            return
+
+        completion_provider = session.get_capability('completionProvider')
+        has_resolve_provider = completion_provider and completion_provider.get('resolveProvider', False)
+        if has_resolve_provider:
+            client.send_request(
+                Request.resolveCompletionItem(item),
+                lambda res: self.handle_resolve_response(res))
+
+    def handle_resolve_response(self, item: Optional[dict]) -> None:
+        if not item:
+            return
+        detail = minihtml(item.get('detail') or "", self.view)
+        documentation = minihtml(item.get("documentation") or "", self.view)
+
+        content = self.get_content(documentation, detail)
+        if self.view.is_popup_visible():
```

Comment:
This will most likely interfere with the signature help state machine if we decide to show sigHelp when the AC is open, but I suppose that is a story for another time.